From 26b8727e49436ddb55a3250973b0c8043f8ecc85 Mon Sep 17 00:00:00 2001
From: Rich Megginson <rmeggins@redhat.com>
Date: Fri, 11 Apr 2014 10:26:35 -0600
Subject: [PATCH 201/225] Ticket #47772 empty modify returns
 LDAP_INVALID_DN_SYNTAX

https://fedorahosted.org/389/ticket/47772
Reviewed by: tbordaz, mreynolds (Thanks!)
Branch: 389-ds-base-1.2.11
Fix Description: tbordaz pointed out a couple of other places where the
pblock modify_mods structure is being referenced directly, and needs to be
checked for NULL.
Platforms tested: RHEL6 x86_64
Flag Day: no
Doc impact: no
(cherry picked from commit 2d25bb4dd39d09d8b5fc5b69702d6e1cccc2c65f)
(cherry picked from commit ba5938bfa94d906c2189ebc6579057b2779ee68d)
(cherry picked from commit 1f3d9f685f2adbb4dc0ebcec3def4ecf68a87013)
(cherry picked from commit 36f51eaf0f57be33741c3736df0d9a802938a816)
---
 ldap/servers/plugins/replication/repl5_protocol_util.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ldap/servers/plugins/replication/repl5_protocol_util.c b/ldap/servers/plugins/replication/repl5_protocol_util.c
index 4449170..2fbe7c4 100644
--- a/ldap/servers/plugins/replication/repl5_protocol_util.c
+++ b/ldap/servers/plugins/replication/repl5_protocol_util.c
@@ -695,6 +695,10 @@ repl5_strip_fractional_mods(Repl_Agmt *agmt, LDAPMod ** mods)
 	int strip = 1;
 	int i, j, k;
 
+	if (mods == NULL) {
+                return retval;
+	}
+
 	if (a) {
 		/* Iterate through the fractional attr list */
 		for ( i = 0; a[i] != NULL; i++ ) 
-- 
1.8.1.4

